CONTRIBUTING.md

```diff
@@ -42,6 +42,7 @@ We have few hard rules, but we have some:
 - Calls to `abort()` are forbidden in the core library. This follows from the [Writing R Extensions](https://cran.r-project.org/doc/manuals/R-exts.html) manual which states that "Under no circumstances should your compiled code ever call abort or exit".
 - All source code files (.h, .cpp) must be ASCII.
 - All C macros introduced in public headers need to be prefixed with either `SIMDJSON_` or `simdjson_`.
+- We avoid trailing white space characters. Please avoid making unnecessary changes to white-space characters.
```

Comment:
I would say that trailing white space and formatting are two separate issues. 
I would like this to be only:
- We avoid trailing white space characters.

I agree with trying to keep out of whitespace changes, but that should be covered by some kind of formatting guideline. I plan to have a look on formatting later. @jkeiser said it had been discussed earlier, there is even .clang-format and a CI job using it already. I want to see what has been discussed before I start digging.

Reply:
@pauldreik See me comment later about how we do not want users to have a particular setup, a particular version of clang-format and so forth... to work on simdjson. In Go, for example, we typically require users to do "go fmt" because there is a standard formatter that everyone who uses Go has... But in C++, there is no such thing. That is, a given team might have a strict style policy and strict style guidelines... 

I guess that the equivalent here would be for CMake to provide style checking. If it did so, then we could/should enable it.